src/components/faq/FaqPage.tsx

```diff
@@ -32,22 +32,31 @@ const FAQ_PAGE_QUESTIONS: Record<string, ContentType[]> = {
 
 export default function FaqPage({ section }: { section: FaqCategory }) {
   const { t } = useTranslation()
-  const [value, setValue] = useState(section)
+  const [selectedFaqCategory, setSelectedFaqCategory] = useState(section)
   const [searchKeyword, setSearchKeyword] = useState('')
 
   const faqQuestionsData = filterFaqQuestions(FAQ_PAGE_QUESTIONS, searchKeyword)
   const faqCategories = Object.keys(faqQuestionsData) as FaqCategory[]
 
+  useEffect(() => {
+    if (faqCategories.length > 0 && !faqCategories.includes(selectedFaqCategory)) {
+      setSelectedFaqCategory(faqCategories[0])
+    }
+  }, [searchKeyword])
```

Comment:
@valkirilov This can be changed to `useMemo` for brevity.
When `useEffect` only uses `setState` we can normally convert it.

See https://beta.reactjs.org/learn/you-might-not-need-an-effect




```suggestion
  const [searchKeyword, setSearchKeyword] = useState('')

  const faqQuestionsData = useMemo(() => filterFaqQuestions(FAQ_PAGE_QUESTIONS, searchKeyword), [searchKeyword])
  const faqCategories = useMemo(() => Object.keys(faqQuestionsData) as FaqCategory[], [faqQuestionsData)

  const selectedFaqCategory = useMemo(() => {
    if (faqCategories.length > 0 && !faqCategories.includes(selectedFaqCategory)) {
      return faqCategories[0]
    }
    return section
  }, [faqCategories, section])
```

Reply:
Great suggestion! The changes are applied, but I had to remove the option to select categories from the Tabs via the state setter (because we remove it from the parent component). It's a bit slower on local, because we rely only on redircet of the page to change the selected category/section, but I guess that it won't be an issue on DEV and PROD:
- rework the `FaqPage` component so we memoize the selected category and the logic which we apply to it when using the search filter (to keep always a selected category when we run out of matches in the current one)
- remove the `setSelectedCategory` from the `VerticalTabs` component and make it so we rely only on the redirect of the page to change the selected category